src/createFirebaseInstance.js

```diff
@@ -240,6 +241,27 @@ export const createFirebaseInstance = (firebase, configs, dispatch) => {
     storageActions.deleteFile(dispatch, firebase, { path, dbPath })
 
   /**
+   * @description Connect. Similar to the firebaseConnect Higher Order
+   * Component but presented as a function. Useful for populating your redux
+   * state without React, e.g., for service side rendering.
+   * @param {Array} watchArray - Array of objects or strings for paths to sync
+   * from Firebase. Can also be a function that returns the array. The function
+   * is passed the props object specified as the next parameter.
+   * @param {Object} props - The props object that you would like to pass to
+   * your watchArray generating function.
+   * @return {Promise}
+   */
+  const connect = (watchArray, props) => {
+    const inputAsFunc = createCallable(watchArray)
+    const prevData = inputAsFunc(props, firebase)
+    const firebaseEvents = getEventsFromInput(prevData)
+    const promises = firebaseEvents.map(event =>
```

Comment:
Kinda a nitpick: We should be able to use `watchEvents` and just pass the listener event settings array right into it instead of needing to call the `map` here.

Reply:
That's what I had initially but watchEvents doesn't return a promise, which is important for SSR as you have to wait for the data to be fetched before rendering. I could update watchEvents if you want.